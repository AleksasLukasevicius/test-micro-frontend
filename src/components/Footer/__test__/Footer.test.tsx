import { render, screen } from "@testing-library/react";
import { Footer } from "../Footer";
import renderer from "react-test-renderer";

describe("Footer", () => {
  it("shoul match snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Footer", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("should render Footer", () => {
    render(<Footer />);
    expect(document.querySelector("footer")).toBeInTheDocument();
  });

  it("should render current year", () => {
    render(<Footer />);

    const currentYearElement = screen.getByLabelText("current year");
    expect(currentYearElement).toBeVisible();
    // expect(currentYearElement.textContent).toBe(new Date().getFullYear().toString())
    expect(currentYearElement.textContent).toBe(`${new Date().getFullYear()}`);
  });

  it("should render current year", () => {
    render(<Footer />);
    expect(document.querySelector("footer")).toHaveTextContent(
      new Date().getFullYear().toString()
    );
  });
});
