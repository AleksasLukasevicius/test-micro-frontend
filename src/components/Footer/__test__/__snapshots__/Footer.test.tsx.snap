// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer shoul match snapshot 1`] = `
<footer>
  <p
    aria-label="current year"
  >
    2023
  </p>
</footer>
`;

exports[`Footer should render Footer 1`] = `
<div>
  <footer>
    <p
      aria-label="current year"
    >
      2023
    </p>
  </footer>
</div>
`;
